pilot/pkg/networking/core/v1alpha3/accesslog.go

```diff
@@ -138,18 +143,66 @@ type AccessLogBuilder struct {
 	mutex                 sync.RWMutex
 	fileAccesslog         *accesslog.AccessLog
 	listenerFileAccessLog *accesslog.AccessLog
+
+	storeMutex sync.RWMutex
+	store      simplelru.LRUCache
 }
 
 func newAccessLogBuilder() *AccessLogBuilder {
-	return &AccessLogBuilder{
+	b := &AccessLogBuilder{
 		tcpGrpcAccessLog:         buildTCPGrpcAccessLog(false),
 		httpGrpcAccessLog:        buildHTTPGrpcAccessLog(),
 		tcpGrpcListenerAccessLog: buildTCPGrpcAccessLog(true),
 	}
+
+	b.store, _ = simplelru.NewLRU(cacheSize, nil)
+
+	return b
+}
+
+func (b *AccessLogBuilder) cachedAccessLog(key interface{}) ([]*accesslog.AccessLog, bool) {
+	b.storeMutex.Lock()
+	defer b.storeMutex.Unlock()
+
+	if !b.store.Contains(key) {
+		return nil, false
+	}
+	val, ok := b.store.Get(key)
+	if !ok {
+		return nil, false
+	}
+
+	logs, ok := val.([]*accesslog.AccessLog)
+	return logs, ok
+}
+
+func (b *AccessLogBuilder) addAccessLog(key interface{}, accessLoggings []*accesslog.AccessLog) {
```

Comment:
nit: addAccessLog -> cacheAccessLog may be better?

Reply:
that will a little confuse with L165 `cachedAccessLog`

Reply:
Ok. This is not clear where it is adding